public/app/features/live/live.ts

```diff
@@ -0,0 +1,113 @@
+import Centrifuge, { PublicationContext } from 'centrifuge/dist/centrifuge.protobuf';
+import SockJS from 'sockjs-client';
+import { GrafanaLiveSrv, setGrafanaLiveSrv, ChannelHandler } from '@grafana/runtime';
+import { PartialObserver, Unsubscribable, Subject, BehaviorSubject } from 'rxjs';
+import { KeyValue } from '@grafana/data';
+
+interface Channel<T = any> {
+  subject: Subject<T>;
+  subscription?: Centrifuge.Subscription;
+}
+
+class CentrifugeSrv implements GrafanaLiveSrv {
+  centrifuge: Centrifuge;
+  channels: KeyValue<Channel> = {};
```

Comment:
Currently this is a *super* naive approach, and should be improved.  In particular, this will hold on to subscriptions after everyone has unsubscribed

Reply:
In this approach plugins do not have direct access to `centrifuge` and this server intercepts all messages and manages rxjs subscriptions.  Alternatively, we could expose centrifuge and give plugins full control/responsibility.

This initial approach feels like the least risky, but we will could always expose centrifuge if it becomes too difficult to wrap everything